### ISSUE TITLE
build fails against msgpack-2.0.0

### ISSUE THREAD

**Issue by hakamadare** [rogual/homebrew-neovim-dot-app#16]
```
$ brew info msgpack
msgpack: stable 2.0.0, HEAD
Library for a binary-based efficient data interchange format
https://msgpack.org/
/opt/boxen/homebrew/Cellar/msgpack/1.4.2 (478 files, 4.7M)
  Poured from bottle on 2016-06-13 at 09:31:43
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/msgpack.rb
==> Dependencies
Build: cmake ✔
```

so far, so good.  let's build Neovim.app.

```
$ brew install --HEAD neovim-dot-app
==> Installing neovim-dot-app from rogual/neovim-dot-app
==> Installing dependencies for rogual/neovim-dot-app/neovim-dot-app: msgpack
==> Installing rogual/neovim-dot-app/neovim-dot-app dependency: msgpack
==> Downloading https://github.com/msgpack/msgpack-c/releases/download/cpp-2.0.0/msgpack-2.0.0.tar.gz
Already downloaded: /opt/boxen/cache/homebrew/msgpack-2.0.0.tar.gz
==> cmake . -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/opt/bo
==> make install
🍺  /opt/boxen/homebrew/Cellar/msgpack/2.0.0: 649 files, 5M, built in 7 seconds
==> Installing rogual/neovim-dot-app/neovim-dot-app
==> Cloning https://github.com/rogual/neovim-dot-app.git
Updating /opt/boxen/cache/homebrew/neovim-dot-app--git
==> Checking out branch master
==> make NVIM=/opt/boxen/homebrew/opt/neovim/bin/nvim
Last 15 lines from /Users/rk/Library/Logs/Homebrew/neovim-dot-app/01.make:
VIM=/opt/boxen/homebrew/Cellar/neovim/0.1.4/share/nvim NVIM=/opt/boxen/homebrew/opt/neovim/bin/nvim scons -Q
Install file: "res/Info.plist" as "build/Neovim.app/Contents/Info.plist"
clang++ -o build/client.o -c -std=c++11 -stdlib=libc++ -g -Wno-deprecated-register -I/opt/boxen/homebrew/Cellar/msgpack/2.0.0/include -Ibuild -Isrc src/client.cc
src/client.cc:48:20: error: no matching member function for call to 'next'
    while(unpacker.next(&unpacked)) {
          ~~~~~~~~~^~~~
/opt/boxen/homebrew/Cellar/msgpack/2.0.0/include/msgpack/v2/unpack.hpp:1335:23: note: candidate function not viable: no known conversion from 'msgpack::unpacked *' (aka 'msgpack::v1::object_handle *') to 'msgpack::object_handle &' for 1st argument; remove &
inline bool unpacker::next(msgpack::object_handle& result) {
                      ^
/opt/boxen/homebrew/Cellar/msgpack/2.0.0/include/msgpack/v2/unpack.hpp:1320:23: note: candidate function not viable: requires 2 arguments, but 1 was provided
inline bool unpacker::next(msgpack::object_handle& result, bool& referenced) {
                      ^
1 error generated.
scons: *** [build/client.o] Error 1
make: *** [all] Error 2

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/brew):
  https://github.com/rogual/homebrew-neovim-dot-app/issues
```

:sadtrombone: neovim-dot-app pulls in the latest msgpack, then fails to build.  i've tried to work around this by pinning msgpack to 1.4.2, but that didn't prevent neovim-dot-app from installing the latest msgpack anyway; the workaround turns out to be `brew switch msgpack 1.4.2 && brew install --HEAD neovim-dot-app`.

here are the [release notes](https://github.com/msgpack/msgpack-c/releases/tag/cpp-2.0.0) for msgpack-2.0.0; i hope something in there is meaningful to you.  i'll try to dig in and figure out what went wrong if i can find the time, but this is really not my area of expertise.

**Comment by hakamadare** [comment]
oh hey, it looks like rogual/neovim-dot-app#272 fixes this, perhaps?

**Comment by lundquistlab** [comment]
I had this same issue, changing those two files from [rougal/neovim-dot-app#272](https://github.com/rogual/neovim-dot-app/pull/272) worked and I was successfully able to compile Neovim.app from source.

**Comment by hakamadare** [comment]
fixed by rogual/neovim-dot-app#272 which is now merged
